src/main/java/org/prebid/server/bidder/tappx/TappxBidder.java

```diff
@@ -107,18 +109,29 @@ private String buildEndpointUrl(ExtImpTappx extImpTappx, Integer test) {
             throw new PreBidException("Tappx tappxkey undefined");
         }
 
-        String url = String.format("%s%s/%s?tappxkey=%s", endpointUrl, host, endpoint, tappxkey);
-        if (test != null && test == 0) {
-            int t = (int) System.nanoTime();
-            url += "&ts=" + t;
-        }
-
-        url += "&v=" + VERSION;
-        url += "&type_cnn=" + TYPE_CNN;
+        return tryBuildUrl(endpointUrl, host, endpoint, tappxkey, test);
+    }
 
+    private String tryBuildUrl(String endpointUrl, String host, String endpoint, String tappxkey, Integer test) {
```

Comment:
Seems we can change some names
`buildEndpointUrl` -> `resolveUrl`
`tryBuildUrl` -> `buildUrl`

Reply:
Refactored